tests/it/query.rs

```diff
@@ -176,3 +176,21 @@ async fn all_floats() {
 
     assert_eq!(vec, &[42.5, 43.5]);
 }
+
+#[tokio::test]
+async fn settings_override() {
+    let client = prepare_database!();
+
+    let setting_name = "date_time_input_format";
+    let setting_value = "basic";
+
+    let value = client
+        .query("SELECT value FROM system.settings WHERE name = ?")
+        .bind(setting_name)
+        .with_option(setting_name, setting_value)
+        .fetch_one::<String>()
+        .await
+        .unwrap();
+
```

Comment:
I think we should also check that query-specific settings don't affect client ones.

Reply:
Added.